src/lib.rs

```diff
@@ -1,10 +1,48 @@
+//! Epoch-based memory reclamation.
+//!
+//! # Pointers
+//!
+//! Concurrent collections are built using atomic pointers. This module provides [`Atomic`], which
+//! is just a shared atomic pointer to a heap-allocated object. Loading an [`Atomic`] yields a
+//! [`Ptr`], which is an epoch-protected pointer through which the loaded object can be safely read.
+//!
+//! # Pinning
+//!
+//! Before an [`Atomic`] can be loaded, the current mutator must be [`pin`]ned. By pinning a mutator
```

Comment:
That is related to @stjepang 's question: will `Mutator` have a lifetime different from `'static`?  (BTW, "mutator" seems an established terminology among garbage collection guys.)  Naming the struct as `Mutator` and allowing other lifetimes than `'static` is an effort for separation of concern.  Files `mutator.rs` and `epoch.rs` concern only with the struct for mutators and the global epoch, respectively.  File `global.rs`'s role is actually defining the global epoch and defining a mutator for each thread.

NB: I came to design after exploring the design space of `Namespace` (now I renamed it to `Realm`).  With that generalization in mind, I needed to separate of the struct (`mutator.rs` and `epoch.rs`, and presumably `garbage.rs` later) and the global instance of realm (`global.rs`).